src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -812,15 +845,26 @@ public static void setSerialFilterFactory(BinaryOperator<ObjectInputFilter> filt
             }
             if (filterFactoryNoReplace.getAndSet(true)) {
                 final String msg = serialFilterFactory != null
-                        ? serialFilterFactory.getClass().getName()
-                        : "initialization incomplete";
-                throw new IllegalStateException("Cannot replace filter factory: " + msg);
+                        ? "Cannot replace filter factory: " + serialFilterFactory.getClass().getName()
+                        : invalidFilterFactoryMessage();
+                throw new IllegalStateException(msg);
             }
             configLog.log(DEBUG,
                     "Setting deserialization filter factory to {0}", filterFactory.getClass().getName());
             serialFilterFactory = filterFactory;
         }
 
+        /*
+         * Return message for an invalid filter factory configuration saved from the static init.
+         * It can be called before the static initializer is complete and has set the message/null.
```

Comment:
More of a curiosity than a review comment - Is that because the `Config.getSerialFilterFactory()/setSerialFilterFactory(...)` could be called from the constructor of the user configured factory class, which gets invoked when static initialization is in progress for the `Config` class?

Reply:
Yes, it can be called before the `Config` class has completed initialization and `invalidFactoryMessage` may be seen as null.